src/Kernel.php

```diff
@@ -107,16 +162,18 @@ function (RouteCollector $routeCollector) use ($collector) {
             }
         ));
 
-        $middlewareQueue[] = $this->di->makeFromDefinition(
-            RequestHandler::class
-        );
+        $middlewareQueue[] = $this->di->get(RequestHandler::class);
+
+        $emitterStack = $this->di->get(EmitterStack::class);
+
+        $emitterStack->push($this->di->get(SapiEmitter::class));
+
+        $emitterStack->push($this->di->get(ConditionalSapiStreamEmitter::class));
```

Comment:
Since `EmitterStack` and the emitters are all configured in this package's DI config, perhaps it would provide more flexibility to push the emitters onto the stack while setting up `EmitterStack` in the config rather than doing it here. That would allow an app that uses this package to override which emitters they want to use and in which order. (Apps can override or extend the DI config, right?)

Reply:
@kevinsmith You make a great point. Right now array_merge is used to pull together all the arrays of DI config. Unfortunately it works the other way because the base project's DI config is the first thing retrieved, then it merges in each of the others, which is simply a complete oversight on my part. It absolutely should work where the project has the final say on DI config. I'll have a stern conversation with myself.

The collector is a separate composer package because I use this methodology in a couple of packages. At some future date, I will implement a change to the order to array_merge the base project's config last, allowing it final say. But it shouldn't affect us for the moment.

As to doing the emitter stack in the DI config, that's a great point and I'll implement it that way.

Reply:
@kevinsmith https://github.com/buzzingpixel/corbomite-http/pull/1/commits/6b9e219a648680908586a6d8ae5ffa037bb9f08e